static/js/src/advantage/users/components/TableView/components.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { format } from "date-fns";
+
+import Button from "@canonical/react-components/dist/components/Button";
+import Select from "@canonical/react-components/dist/components/Select";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+
+import { User } from "../../types";
+import { UserRowVariant } from "./TableView";
+
+const DATE_FORMAT = "dd/MM/yyyy";
+
+type UserVariantProps = {
+  user: User;
+  variant: UserRowVariant;
+};
+
+type UserActionsProps = {
+  handleEditOpen: (id: string) => void;
+  handleEditSubmit: (id: string) => void;
+  handleCancel: () => void;
+} & UserVariantProps;
+
+const UserActions = ({
+  user,
+  variant,
+  handleEditOpen,
+  handleEditSubmit,
+  handleCancel,
+}: UserActionsProps) => {
+  return variant !== "editing" ? (
+    <Button
+      small
+      dense
+      onClick={() => handleEditOpen(user.id)}
+      disabled={variant === "disabled"}
+      aria-label={`Edit user ${user.email}`}
+    >
+      Edit
+    </Button>
+  ) : (
+    <>
+      <Button small dense onClick={handleCancel}>
+        Cancel
+      </Button>
+      <Button
+        small
+        dense
+        appearance="positive"
+        onClick={() => handleEditSubmit(user.id)}
+      >
+        Save
+      </Button>
```

Comment:
You might want to use the [ActionButton](https://canonical-web-and-design.github.io/react-components/?path=/docs/actionbutton--default-story) here so you can use its loading state while you wait for a response from the backend